### ISSUE TITLE
CTRL-C gives traceback on py3 setup

### ISSUE THREAD

**Issue by frogunder** [saltstack/salt#45956]
### Description of Issue/Question
Running command and hitting CTRL-C before it is completed exits with a traceback on a py3 setup in oxygen.rc1.

Should see something like this:
```
[root@localhost ~]# salt '*' test.sleep 30
^C
Exiting gracefully on Ctrl-c
This job's jid is: 20180209183728441275
The minions may not have all finished running and any remaining minions will return upon completion. To look up the return data for this job later, run the following command:

salt-run jobs.lookup_jid 20180209183728441275
[root@localhost ~]# 
```

But see this:
```
[root@localhost ~]# salt '*' test.sleep 30
^C[ERROR   ] An un-handled exception was caught by salt's global exception handler:
AttributeError: 'NoneType' object has no attribute '__context__'
Traceback (most recent call last):
  File "/usr/bin/salt", line 6, in <module>
    exec(compile(open(__file__).read(), __file__, 'exec'))
  File "/root/salt/scripts/salt", line 10, in <module>
    salt_main()
  File "/root/salt/salt/scripts.py", line 480, in salt_main
    client.run()
  File "/root/salt/salt/cli/salt.py", line 191, in run
    for full_ret in cmd_func(**kwargs):
  File "/root/salt/salt/client/__init__.py", line 817, in cmd_cli
    **kwargs):
  File "/root/salt/salt/client/__init__.py", line 1608, in get_cli_event_returns
    **kwargs
  File "/root/salt/salt/client/__init__.py", line 1189, in get_iter_returns
    for raw in ret_iter:
  File "/root/salt/salt/client/__init__.py", line 1114, in get_returns_no_block
    no_block=True, auto_reconnect=self.auto_reconnect)
  File "/root/salt/salt/utils/event.py", line 634, in get_event
    ret = self._get_event(wait, tag, match_func, no_block)
  File "/root/salt/salt/utils/event.py", line 541, in _get_event
    raw = self.subscriber.read_sync(timeout=wait)
  File "/root/salt/salt/transport/ipc.py", line 699, in read_sync
    self.io_loop.start()
  File "/usr/lib64/python3.4/site-packages/tornado/ioloop.py", line 863, in start
    event_pairs = self._impl.poll(poll_timeout)
  File "/root/salt/salt/scripts.py", line 46, in _handle_signals
    trace = traceback.format_exc()
  File "/usr/lib64/python3.4/traceback.py", line 256, in format_exc
    return "".join(format_exception(*sys.exc_info(), limit=limit, chain=chain))
  File "/usr/lib64/python3.4/traceback.py", line 181, in format_exception
    return list(_format_exception_iter(etype, value, tb, limit, chain))
  File "/usr/lib64/python3.4/traceback.py", line 146, in _format_exception_iter
    for value, tb in values:
  File "/usr/lib64/python3.4/traceback.py", line 125, in _iter_chain
    context = exc.__context__
AttributeError: 'NoneType' object has no attribute '__context__'
Traceback (most recent call last):
  File "/usr/bin/salt", line 6, in <module>
    exec(compile(open(__file__).read(), __file__, 'exec'))
  File "/root/salt/scripts/salt", line 10, in <module>
    salt_main()
  File "/root/salt/salt/scripts.py", line 480, in salt_main
    client.run()
  File "/root/salt/salt/cli/salt.py", line 191, in run
    for full_ret in cmd_func(**kwargs):
  File "/root/salt/salt/client/__init__.py", line 817, in cmd_cli
    **kwargs):
  File "/root/salt/salt/client/__init__.py", line 1608, in get_cli_event_returns
    **kwargs
  File "/root/salt/salt/client/__init__.py", line 1189, in get_iter_returns
    for raw in ret_iter:
  File "/root/salt/salt/client/__init__.py", line 1114, in get_returns_no_block
    no_block=True, auto_reconnect=self.auto_reconnect)
  File "/root/salt/salt/utils/event.py", line 634, in get_event
    ret = self._get_event(wait, tag, match_func, no_block)
  File "/root/salt/salt/utils/event.py", line 541, in _get_event
    raw = self.subscriber.read_sync(timeout=wait)
  File "/root/salt/salt/transport/ipc.py", line 699, in read_sync
    self.io_loop.start()
  File "/usr/lib64/python3.4/site-packages/tornado/ioloop.py", line 863, in start
    event_pairs = self._impl.poll(poll_timeout)
  File "/root/salt/salt/scripts.py", line 46, in _handle_signals
    trace = traceback.format_exc()
  File "/usr/lib64/python3.4/traceback.py", line 256, in format_exc
    return "".join(format_exception(*sys.exc_info(), limit=limit, chain=chain))
  File "/usr/lib64/python3.4/traceback.py", line 181, in format_exception
    return list(_format_exception_iter(etype, value, tb, limit, chain))
  File "/usr/lib64/python3.4/traceback.py", line 146, in _format_exception_iter
    for value, tb in values:
  File "/usr/lib64/python3.4/traceback.py", line 125, in _iter_chain
    context = exc.__context__
AttributeError: 'NoneType' object has no attribute '__context__'
[root@localhost ~]# 
```

### Versions Report
```
Salt Version:
           Salt: 2018.2.0-2269-g78fed2d
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
      docker-py: Not Installed
          gitdb: 2.0.3
      gitpython: 2.1.8
          ioflo: Not Installed
         Jinja2: 2.10
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.5.4
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 3.4.5 (default, Dec 11 2017, 14:22:24)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 16.0.4
           RAET: Not Installed
          smmap: 2.0.3
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.1.6
 
System Versions:
           dist: centos 7.4.1708 Core
         locale: UTF-8
        machine: x86_64
        release: 3.10.0-693.el7.x86_64
         system: Linux
        version: CentOS Linux 7.4.1708 Core
```

**Comment by rallytime** [comment]
@DmitryKuzmenko Can you look into this one?

**Comment by DmitryKuzmenko** [comment]
👍

**Comment by DmitryKuzmenko** [comment]
@rallytime 👍

**Comment by rallytime** [comment]
Thanks!

**Comment by DmitryKuzmenko** [comment]
Can't reproduce on Arch. Trying in Cent7 container.
Btw, @frogunder are you sure it's `oxygen.rc1`? In the versions report I see `g78fed2d`, but I see that revision is included to `develop` only.
```
$ git branch --all --contains 78fed2d
  remotes/ss/develop
```

**Comment by frogunder** [comment]
Tried again on the head of oxygen.rc1. I still see it on a Cent7 vm but on Arch vm it seems to work correctly.

**Comment by DmitryKuzmenko** [comment]
Ok, thank you for checking.

**Comment by DmitryKuzmenko** [comment]
Reproduced on CentOS and then on Arch Linux. It looks the only sensible thing is the Python version. It's reproducible on Python 3.4 (3.4.5 on Cent and 3.4.7 Arch AUR). Investigating.

**Comment by rallytime** [comment]
@frogunder This should be fixed with #46032
